frontend/public/components/utils/list-dropdown.jsx

```diff
@@ -175,22 +194,77 @@ ListDropdown.propTypes = {
   id: PropTypes.string,
 };
 
-const NsDropdown_ = (props) => {
-  const openshiftFlag = props.flags[FLAGS.OPENSHIFT];
-  if (flagPending(openshiftFlag)) {
-    return null;
+const useProjectOrNamespaceModel = () => {
+  const canCreateNamespace = useAccessReview({
+    group: NamespaceModel.apiGroup,
+    resource: NamespaceModel.plural,
+    verb: 'create',
+  });
+
+  const canCreateProject = useFlag(FLAGS.CAN_CREATE_PROJECT);
+  const openshiftFlag = useFlag(FLAGS.OPENSHIFT);
+
+  if (flagPending(openshiftFlag) || flagPending(canCreateProject)) {
+    return [];
   }
-  const kind = openshiftFlag ? 'Project' : 'Namespace';
-  const resources = [{ kind }];
-  return (
+
+  // NamespaceModal is used when not on an openshift cluster
+  const model = canCreateNamespace || !openshiftFlag ? NamespaceModel : ProjectModel;
+  const canCreate = canCreateNamespace || (openshiftFlag && canCreateProject);
+  return [model, canCreate];
+};
+
+/** @type {React.FC<{dataFilter?: (ns: any) => boolean, desc?: string, selectedKey?: string, selectedKeyKind?: string, fixed?: boolean, placeholder?: string, onChange?: (selectedKey: string, event: React.Event) => void, id?: string, dataTest?: string}}>} */
+export const NsDropdown = (props) => {
+  const [selectedKey, setSelectedKey] = React.useState('');
```

Comment:
One more small change. This ensures the initial selection is correct:
```suggestion
  const [selectedKey, setSelectedKey] = React.useState(props.selectedKey);
```